### DISCUSSION TITLE
通过socks协议让不同软件走不同线路的问题

### DISCUSSION THREAD

**Discussion by charure** [v2fly/v2ray-core#509] [Q&A]
各位好，
我现在已经实现了不同的软件通过不同的端口走不同的线路，但是看起来和缝合怪似的，
具体如下：

服务端：我目前有多个服务器，均架设好了服务端，这点没问题。

然后在本地，我创建了不同的文件夹，每个文件夹中放一个v2ray，
每个客户端，config里设置不同的端口 比如1号文件夹里监听的是 1081，二号1082，以此类推

最后在软件里，代理选择socks，然后分别填入不同的端口。

用是可以用，但想精简下，能不能在一个config放下所有的配置，实现不同的端口使用不同的线路。
只需要无脑复制进一个config.json中就能实现了吗？

斗胆请教下该怎么优化配置信息，避免看起来过于杂乱

谢谢

**Comment by it-currently-requires-a-github-username** [comment]
只需要无脑复制进一个 config.json 中，加好对应的 tag 就能实现了

**Reply by charure** [reply]
无脑复制至一个config中只有第一个能用 
我想肯定是关于“加好对应tag”这里我的理解出了问题，
麻烦下能给个例子吗？
下面两个 假设一个端口是1081 一个端口是1082
服务器是server1和server2
以此类推
{
  "inbounds": [{
    "port": 1080,  // SOCKS 代理端口，在浏览器中需配置代理并指向这个端口
    "listen": "127.0.0.1",
    "protocol": "socks",
    "settings": {
      "udp": true
    }
  }],
  "outbounds": [{
    "protocol": "vmess",
    "settings": {
      "vnext": [{
        "address": "server1", // 服务器地址，请修改为你自己的服务器 ip 或域名
        "port": 10086,  // 服务器端口
        "users": [{ "id": "b831381d-6324-4d53-ad4f-8cda48b30811" }]
      }]
    }
  },{
    "protocol": "freedom",
    "tag": "direct",
    "settings": {}
  }],
  "routing": {
    "domainStrategy": "IPOnDemand",
    "rules": [{
      "type": "field",
      "ip": ["geoip:private"],
      "outboundTag": "direct"
    }]
  }
}

{
  "inbounds": [{
    "port": 1081,  // SOCKS 代理端口，在浏览器中需配置代理并指向这个端口
    "listen": "127.0.0.1",
    "protocol": "socks",
    "settings": {
      "udp": true
    }
  }],
  "outbounds": [{
    "protocol": "vmess",
    "settings": {
      "vnext": [{
        "address": "server2", // 服务器地址，请修改为你自己的服务器 ip 或域名
        "port": 10086,  // 服务器端口
        "users": [{ "id": "b831381d-6324-4d53-ad4f-8cda48b30812" }]
      }]
    }
  },{
    "protocol": "freedom",
    "tag": "direct",
    "settings": {}
  }],
  "routing": {
    "domainStrategy": "IPOnDemand",
    "rules": [{
      "type": "field",
      "ip": ["geoip:private"],
      "outboundTag": "direct"
    }]
  }
}

**Reply by charure** [reply]
{
"tag": "program1",
"inbounds": [{
"port": 1080, // SOCKS 代理端口，在浏览器中需配置代理并指向这个端口
"listen": "127.0.0.1",
"protocol": "socks",
"settings": {
"udp": true
}
}],
"outbounds": [{
"protocol": "vmess",
"settings": {
"vnext": [{
"address": "server1", // 服务器地址，请修改为你自己的服务器 ip 或域名
"port": 10086, // 服务器端口
"users": [{ "id": "b831381d-6324-4d53-ad4f-8cda48b30811" }]
}]
}
},{
"protocol": "freedom",
"tag": "direct",
"settings": {}
}],
"routing": {
"domainStrategy": "IPOnDemand",
"rules": [{
"type": "field",
"ip": ["geoip:private"],
"outboundTag": "direct"
}]
}
}

**Comment by kslr** [comment]
你可以试试v5版的多配置，只需要复制进去一份单独的配置
